src/geometry/manifold/tangent_bundle.lean

```diff
@@ -455,28 +447,30 @@ def tangent_bundle_core : basic_smooth_bundle_core I M E :=
         (is_open.mem_nhds (local_homeomorph.open_source _) _),
       simpa only [model_with_corners.left_inv] using hx },
     rw [B, C, D, E] at A,
-    simp only [A, continuous_linear_map.coe_comp'] with mfld_simps
+    simp only [A, continuous_linear_map.coe_comp'] with mfld_simps,
+    simp only [forall_const, eq_self_iff_true, linear_map.coe_comp, continuous_linear_map.coe_comp],
   end }
 
 variable {M}
 include I
 
 /-- The tangent space at a point of the manifold `M`. It is just `E`. We could use instead
-`(tangent_bundle_core I M).to_topological_fiber_bundle_core.fiber x`, but we use `E` to help the
+`(tangent_bundle_core I M).to_topological_vector_bundle_core.fiber x`, but we use `E` to help the
 kernel.
 -/
-@[nolint unused_arguments]
-def tangent_space (x : M) : Type* := E
+@[nolint unused_arguments, reducible]
+def tangent_space (x : M) : Type* :=
+(tangent_bundle_core I M).to_topological_vector_bundle_core.fiber x
 
 omit I
 variable (M)
 
 /-- The tangent bundle to a smooth manifold, as a plain type. We could use
-`(tangent_bundle_core I M).to_topological_fiber_bundle_core.total_space`, but instead we use the
+`(tangent_bundle_core I M).to_topological_vector_bundle_core.total_space`, but instead we use the
 (definitionally equal) `Σ (x : M), tangent_space I x`, to make sure that rcasing an element of the
 tangent bundle gives a second component in the tangent space. -/
 @[nolint has_inhabited_instance, reducible] -- is empty if the base manifold is empty
-def tangent_bundle := Σ (x : M), tangent_space I x
+def tangent_bundle := (tangent_bundle_core I M).to_topological_vector_bundle_core.total_space
```

Comment:
Also to get better performance, can you make this the total space of the vector bundle with fiber `tangent_space` (but without unfolding `tangent_space`)?

Reply:
Can you say what you mean with some code? If you mean like `bundle.total_space (tangent_space I)` that does not allow you to use the lemmas from `tangent_bundle_core` so it was not possible to keep it that way

Reply:
Yes, `bundle.total_space (lambda (x : M), tangent_space I x)`. And registering the instances you need on it in a section where you make `tangent_space` reducible, as above.